### PR TITLE
[fix] fully check y array data in ensemble classifiers

### PR DESCRIPTION
**Pull request by icfaust**
### Description 
_Add a comprehensive description of proposed changes_

For sklearn's check_X_y and BaseEnsembles's _validate_data, when multi_output is set to False, check_array is not run on the y input. This will cause issues in converting pandas DataFrames and Series to array-like formats, and will prevent the conversion to supported data types (float64 and float32).  This impacts the results of RandomForestClassifier.fit and ExtraTreesClassifier.fit, which will be unable to discern the type of target for classification.  Will fix https://github.com/intel/scikit-learn-intelex/issues/1229

Fixes # - _issue number(s) if exists_

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
- [ ] The unit tests pass successfully.
- [ ] I have run it locally and tested the changes extensively.
- [ ] I have resolved any merge conflicts that might occur with the base branch.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
- [ ] I have added a respective label(s) to PR if I have a permission for that.

### CODE REVIEW

**Comment by icfaust** [comment]
private CI run: http://intel-ci.intel.com/ef436adc-a6d5-f18d-8f2d-a4bf010d0e2e

**Comment by md-shafiul-alam** [comment]
Do we support multi-output (multi-label classifier or multi-target regression) in general for these estimators?

**Comment by icfaust** [comment]
> Do we support multi-output (multi-label classifier or multi-target regression) in general for these estimators?

Nope, its a weirdness in sklearn's check_X_y in order to get check_array to be used for the y array (and is the source of the error). https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/utils/validation.py#L1321

**Review by ethanglaser — APPROVED** [review]
